internal/pkg/template/templates/workloads/services/static-site/cf.yml

```diff
@@ -180,25 +180,30 @@ Resources:
     Metadata:
       aws:copilot:description: A custom resource to trigger the file-copying state machine
     Type: Custom::TriggerStateMachine
-    DependsOn: 
-      - CopyAssetsStateMachine {{- /* This is a real dependency */}}
+    DependsOn:
       - Bucket
       - BucketPolicyForCloudFront
       - CloudFrontOriginAccessControl
       - CloudFrontDistribution
       - TriggerStateMachineFunction
       - TriggerStateMachineFunctionRole
-      - CopyAssetsStateMachine
+      - CopyAssetsStateMachine {{- /* This is a real dependency */}}
       - CopyAssetsStateMachineRole
-      - CopyAssetsFunction
-      - CopyAssetsFunctionRole
       - EnvManagerS3Access
+      {{- if .StaticSiteAlias}}
+      - CustomDomainAction
```

Comment:
we should prolly add a test maybe after the release, that look through all resources in the template, and check if every one of them is in the `DependsOn` list.